code/ship/ship.cpp

```diff
@@ -1865,7 +1865,10 @@ static int parse_ship(const char *filename, bool replace)
 
 	rtn = parse_ship_values(sip, false, first_time, replace);
 
-	return rtn;	//0 for success
+	if (rtn == 0)	//0 for success
+		return ship_id;
+	else
+		return rtn;
```

Comment:
I did this to make it work the same way as `parse_weapon`.

Reply:
But if the return value isn't used anywhere, why is that behavior change desirable or in-scope for this PR?

Reply:
I wanted to improve the consistency between parsing ships and weapons.  Most of the consistency updates (such as they are) will be in the +remove PR, but I made the change here because I was thinking about it at the time.

Reply:
And no, the index isn't actually used, but if we ever needed the functionality in the future, it's a small step to add it now whereas it might be trickier to figure it out later.

Reply:
Okay, but weapon parsing returns the index because countermeasure parsing *uses* the index; what, exactly, would you use the parsed ship index *for*?

Reply:
That's what I mean by "if we ever needed the functionality in the future".

Reply:
If we ever needed the functionality in the future, we could add it in the future (and quite simply); in terms of consistency, I'd much rather change weapon parsing to *not* need to return the index than superfluously add it to ship parsing.